adaptive_components/lib/src/adaptive_column.dart

```diff
@@ -0,0 +1,156 @@
+            minWidth: _entry.adaptiveWindowType.widthRangeValues.start,
+            maxWidth: _entry.adaptiveWindowType.widthRangeValues.end,
+          ),
+          child: Wrap(
+            runSpacing: 8.0,
+            children: () {
+              int currentColumns = 0;
+              int totalGutters = 0;
+              List<Widget> childs = [];
+              List<AdaptiveContaine> row = [];
+
+              for (AdaptiveContaine child in children) {
+                // The if statement checks if the adaptiveContainer child fits
+                // within the adaptive constraints.
+                if (child.adaptiveConstraints
+                    .withinAdaptiveConstraint(context)) {
+                  row.add(child);
+                  currentColumns += child.adaptiveColumn;
+
+                  if (currentColumns < _totalColumns) {
+                    totalGutters++;
+                  } else {
+                    int gutters = 0;
+                    if (currentColumns > _totalColumns) {
+                      totalGutters--;
+                    }
+
+                    for (AdaptiveContaine r in row) {
+                      // maxWidth equals column width without margin and gutters
+                      // divided by the total number of adaptive containers.
+                      double maxWidth = (MediaQuery.of(context).size.width -
+                          _margin * 2 -
+                          _gutter *
+                              (gutters == totalGutters && currentColumns > _totalColumns ? 0 : totalGutters)) /
+                          _totalColumns *
+                          r.adaptiveColumn;
+                      childs.add(
+                        ConstrainedBox(
+                          constraints: BoxConstraints(
+                            minWidth: maxWidth,
+                            maxWidth: maxWidth,
+                          ),
+                          child: r,
+                        ),
+                      );
+
+                      if (gutters < totalGutters && 1 < row.length) {
+                        childs.add(
+                          SizedBox(
+                            width: _gutter,
+                            child: Container(
+                              color: Colors.grey,
+                            ),
+                          ),
+                        );
+                        gutters++;
+                      }
+                    }
+                    totalGutters = 0;
+                    currentColumns = 0;
+                    row = [];
+                  }
+                }
+              }
+              return childs;
+            }(),
+          ),
+        );
+      },
+    );
+  }
+}
+
+/// Used to see if a range of [AdaptiveWindowType] should be shown in the window.
+/// If the user sets one of the variables below to true than that window type
+/// should be shown within the [AdaptiveContainer].
+class AdaptiveConstraintsColumn {
```

Comment:
I'm confused about what this class does. I think its name is a bit unclear since I would think this is a special kind of `Column` widget but it seems to instead just be a type of data structure. I suggest renaming it and updating its documentation

Reply:
Still a bit confused on what this class does.